src/autopas/tuning/tuningStrategy/ruleBasedTuning/RuleBasedTuning.cpp

```diff
@@ -63,12 +71,52 @@ void RuleBasedTuning::reset(size_t iteration, size_t tuningPhase, std::vector<Co
   _removedConfigurations.clear();
   _rulesTooHarsh = false;
   optimizeSuggestions(configQueue, evidenceCollection);
+#endif
 }
 
 long RuleBasedTuning::getLifetimeWouldHaveSkippedTuningTime() const { return _wouldHaveSkippedTuningTimeLifetime; }
 
 long RuleBasedTuning::getLifetimeTuningTime() const { return _tuningTimeLifetime; }
 
+void RuleBasedTuning::optimizeSuggestions(std::vector<Configuration> &configQueue,
+                                          const EvidenceCollection &evidenceCollection) {
+#ifdef AUTOPAS_ENABLE_RULES_BASED_TUNING
```

Comment:
any reason you changed the order of the functions?

Reply:
Yes. To avoid having more `#ifdef`s than needed. So, all of the functions that are not part of the interface are completely wrapped in a single `#ifdef`. But this required moving a couple of functions about.